qchem/pennylane_qchem/qchem.py

```diff
@@ -561,6 +561,87 @@ def generate_hamiltonian(
     return convert_hamiltonian(h_of), nr_qubits
 
 
+def sd_excitations(n_electrons, n_orbitals, delta_sz):
+    r"""Generates single and double excitations from a Hartree-Fock (HF) reference state.
+
+    The singly- and doubly-excited configurations are generated by acting with the operators
+    :math:`\hat T_1` and :math:`\hat T_2` on the HF state:
+
+    .. math: 
+        && \vert \Phi_\mathrm{S} \rangle = \hat{T}_1 \vert \mathrm{HF} \rangle = \sum_{r \in
+        \mathrm{occ} \\ p \in \mathrm{virt}} \hat{c}_p^\dagger \hat{c}_r \vert \mathrm{HF} 
+        \rangle \\ 
+        && \vert \Phi_\mathrm{D} \rangle = \hat{T}_2 \vert \mathrm{HF} \rangle = \sum_{r>s \in 
+        \mathrm{occ} \\ p>q \in \mathrm{virt}} \hat{c}_p^\dagger \hat{c}_q^\dagger 
+        \hat{c}_r \hat{c}_s \vert \mathrm{HF} \rangle
+
+	where the indices :math:`r, s` and :math:`p, q` run over the occupied (occ) and unoccupied, 
+	referred to as virtual (virt), molecular orbitals and :math:`\hat c` and 
+	:math:`\hat c^\dagger` are the electron annihilation and creation operators, respectively. 
+
+    **Example**
+
+    >>> ph, pphh = sd_configs(2, 4, 0)
+    >>> print(ph)
+    [[0, 2], [1, 3]]
+    >>> print(pphh)
+    [[0, 1, 2, 3]]
+
+    Args: 
+        n_electrons (int): number of active electrons 
+        n_orbitals (int): number of active orbitals
+        delta_sz (int): Spin-projection selection rule. 
+            For single excitations ``sz[p] - sz[r] = delta_sz``.
+            For double excitations ``sz[p] + sz[p] - sz[r] - sz[s] = delta_sz``.
+            ``sz`` is the single-particle state spin quantum number and ``delta_sz``, in the 
+            case of singles and doubles, can take the values :math:`0`, :math:`\pm 1` 
+            and :math:`\pm 2`.
+
+    Returns:
+        tuple(list, list): lists with the indices of the molecular orbitals
+        involved in the single and double excitations
+    """
+
+    if not n_electrons > 0:
+        raise ValueError(
+            "The number of active electrons has to be greater than 0 \n"
+            "Got n_electrons = {}".format(n_electrons)
+        )
+
+    if n_orbitals <= n_electrons:
+        raise ValueError(
+            "The number of active orbitals ({}) "
+            "has to be greater than the number of active electrons ({})."
+            .format(n_orbitals, n_electrons)
+        )
+
+    if int(delta_sz) not in (0, 1, -1, 2, -2):
```

Comment:
Is the int necessary here?

Reply:
Thanks @soranjh for this point. It is not necessary. Indeed, having the ``int`` breaks this check:
```
if int(delta_sz) not in (0, 1, -1, 2, -2):
        raise ValueError(
            "Expected values for 'delta_sz' are 0, +/- 1 and +/- 2 but got ({})."
             .format(delta_sz)
        )
```
for, say, ``delta_sz=1.5``, which is not OK. Good catch!.